samplesheets/views_api.py

```diff
@@ -310,6 +323,240 @@ def post(self, request, *args, **kwargs):
         return Response(ret_data, status=status.HTTP_200_OK)
 
 
+class IrodsAccessTicketListAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, APIView
+):
+    """
+    List iRODS access tickets for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/list/{Project.sodar_uuid}``
+
+    **Methods:** ``GET``
+
+    **Query parameters:** ``active`` (boolean, default ``0``)
+
+    **Returns:**
+
+    - ``count``: Number of iRODS access tickets
+    - ``tickets``: List of iRODS access tickets
+    """
+
+    http_method_names = ['get']
+    permission_required = 'samplesheets.edit_sheet'
+
+    def get(self, request, *args, **kwargs):
+        """GET request for listing iRODS access tickets"""
+        project = self.get_project()
+        tickets = IrodsAccessTicket.objects.filter(
+            study__investigation__project=project
+        )
+        active = request.query_params.get('active', '0')
+        active = bool(int(active))
+        if active:
+            tickets = [t for t in tickets if t.is_active()]
+        if tickets:
+            serializer = IrodsAccessTicketSerializer(tickets, many=True)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response(
+                {'detail': IRODS_TICKETS_NOT_FOUND_MSG},
```

Comment:
We should not return any customized return data unless we have some specific results to report to the user. Here we either find the objects or not, so the default DRF 404 response is perfectly fine.

Of course, once we refactor this class into just using `get_queryset()`, we will not have to worry about the actual response anyway :)